pkg/datastore/ingest.go

```diff
@@ -105,6 +105,14 @@ func (s *Datastore) Ingest(ctx context.Context, datasetID string, content []byte
 		return nil, nil
 	}
 
+	// Before adding doc, we need to remove the existing documents for duplicates or old contents
+	where := map[string]string{
+		"absPath": opts.FileMetadata.AbsolutePath,
+	}
+	if err := s.Vectorstore.RemoveDocument(ctx, "", datasetID, where, nil); err != nil {
```

Comment:
The problem here is that if the `AddDocuments` operation fails, the file content will be removed. At this point user will have to re-run the ingestion again to recover.

Reply:
I think this would be expected and reasonable